src/ISBootloaderThread.cpp

```diff
@@ -332,6 +332,20 @@ void cISBootloaderThread::update_thread_libusb(void* context)
     m_libusb_thread_mutex.unlock();
 }
 
+#define RETURN_CHECK_IF_CANCELLED   if(m_uploadProgress(NULL, 0.0f) == IS_OP_CANCELLED) { m_continue_update = false; m_update_in_progress = false; m_update_mutex.unlock(); if(m_waitAction) m_waitAction(); return vector<confirm_bootload_t>(); }
+#define RETURN_IF_CANCELLED         if(m_uploadProgress(NULL, 0.0f) == IS_OP_CANCELLED) { m_continue_update = false; m_update_in_progress = false; m_update_mutex.unlock(); if(m_waitAction) m_waitAction(); return IS_OP_CANCELLED; }
```

Comment:
Similar here.  I see why you're using them, but they feel a bit hackish.  I'm not entirely sure what is happening here with the result values though, so maybe it makes sense.